sda/cmd/s3inbox/proxy.go

```diff
@@ -624,3 +631,28 @@ func reportError(errorCode int, message string, w http.ResponseWriter) {
 		log.Error(err)
 	}
 }
+
+func (p *Proxy) storeObjectSizeInDB(path, fileID string) error {
+	client, err := storage.NewS3Client(p.s3)
+	if err != nil {
+		return err
+	}
+
+	o, err := client.HeadObject(context.TODO(), &s3.HeadObjectInput{
+		Bucket: &p.s3.Bucket,
+		Key:    &path,
+	})
+	if err != nil {
+		return err
+	}
+
+	if err := p.database.DB.Ping(); err != nil {
+		p.database.Close()
+		_ = p.database.Connect()
```

Comment:
How if re-connection is also failed, shouldn't it return the error instead of process further?

Reply:
Not necessary since the call below will fail and then the function will return an error